api/client.go

```diff
@@ -52,6 +54,16 @@ func (cfg *Config) roundTripper() http.RoundTripper {
 	return cfg.RoundTripper
 }
 
+func (cfg *Config) client() *http.Client {
```

Comment:
I would rather prefer to have the same pattern we used for the `RounTripper`. It should be passed as part of configuration, if it doesn't present we can use the default. Also, we need to decide and document what happens when both `RoundTripper` and `Client` used at the same time. I'm in favor of returning an error.